[Upgrade Assistant] Add A11y Tests

cuff-links opened the pull request:
Closes #80059

This PR adds accessibility functional tests for the Upgrade Assistant app.

It covers:

- The overview tab
- The cluster tab
- The indice tab

elasticmachine commented:
Pinging @elastic/kibana-accessibility (Project:Accessibility)

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

cuff-links commented:
@elasticmachine merge upstream

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/)
* Commit: f27f49cd1a7aaa11f132b8695fb01b0a35e93d55
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/execution/node/238/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/","number":"103749","commit":"f27f49cd1a7aaa11f132b8695fb01b0a35e93d55"}]}
PIPELINE-->

alisonelizabeth commented on x-pack/test/accessibility/apps/upgrade_assistant.ts:
```diff
@@ -0,0 +1,36 @@
+
+    it('Overview', async () => {
+      await retry.waitFor('Upgrade Assistant overview tab to be visible', async () => {
+        return testSubjects.exists('upgradeAssistantOverviewTabDetail') ? true : false;
```
Is this ternary necessary? I thought `exists` returns a boolean.

cuff-links replied:
Ah. Redundant. Thanks.

alisonelizabeth commented on x-pack/test/accessibility/apps/upgrade_assistant.ts:
```diff
@@ -0,0 +1,36 @@
+    it('List View', async () => {
+      await testSubjects.click('upgradeAssistantClusterTab');
+      await retry.waitFor('Upgrade Assistant Cluster tab to be visible', async () => {
+        return testSubjects.exists('upgradeAssistantOverviewTab') ? true : false;
```
same question here

cuff-links replied:
Fixed

alisonelizabeth commented on x-pack/test/accessibility/apps/upgrade_assistant.ts:
```diff
@@ -0,0 +1,36 @@
+      await retry.waitFor('Upgrade Assistant Cluster tab to be visible', async () => {
+        return testSubjects.exists('upgradeAssistantOverviewTab') ? true : false;
+      });
+      return testSubjects.isDisplayed('pipelineDetails') ? true : false;
```
Was this line intentional? I'm surprised the test is passing with it.

cuff-links replied:
Fixed

alisonelizabeth commented on x-pack/test/accessibility/apps/upgrade_assistant.ts:
```diff
@@ -0,0 +1,36 @@
+    });
+
+    it('List View', async () => {
+      await testSubjects.click('upgradeAssistantClusterTab');
```
should we also test the indices tab?

alisonelizabeth reviewed (changes requested):
Thanks for working on this @cuff-links! It's great to see more accessibility tests.

I left a few questions in the code. I also think you will need to add the test file to `x-pack/test/accessibility/config.ts` in order for the tests to run in CI.

```
require.resolve('./apps/upgrade_assistant'),
```

alisonelizabeth commented:
I just noticed CI failures and I think there are a couple other things that will need to be addressed:

- `x-pack/test/accessibility/apps/upgrade_assistant.ts` will need to be updated with the new license header
- The jest snapshot for `checkup_tab.test.tsx` will need to be updated to account for the `data-test-subj` that was added

cjcenizal commented:
@cuff-links This PR needs the 7.12.0 version label if you're planning on backporting to 7.x for the 7.12.0 release.

cuff-links commented:
@alisonelizabeth This PR wasn't ready yet. I was just pushing some commits. It wasn't supposed to flag you at all. Sorry about that. I am addressing the feedback you left. I was just pushing this more for tracking purposes.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/)
* Commit: 7e05b927382e5ccfd03854e16b1844f243c951cd
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/execution/node/238/log)

### History
* :broken_heart: [Build #103749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/) failed f27f49cd1a7aaa11f132b8695fb01b0a35e93d55

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/","number":"103749","commit":"f27f49cd1a7aaa11f132b8695fb01b0a35e93d55"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/","number":"104111","commit":"7e05b927382e5ccfd03854e16b1844f243c951cd"}]}
PIPELINE-->

cuff-links commented:
@elasticmachine merge upstream

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104127/)
* Commit: ea9e086c9aa36ce3ca258e96e1d3e507f5d503ba
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104127/execution/node/238/log)

### History
* :broken_heart: [Build #104111](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/) failed 7e05b927382e5ccfd03854e16b1844f243c951cd
* :broken_heart: [Build #103749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/) failed f27f49cd1a7aaa11f132b8695fb01b0a35e93d55

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/","number":"103749","commit":"f27f49cd1a7aaa11f132b8695fb01b0a35e93d55"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/","number":"104111","commit":"7e05b927382e5ccfd03854e16b1844f243c951cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104127/","number":"104127","commit":"ea9e086c9aa36ce3ca258e96e1d3e507f5d503ba"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104154/)
* Commit: ac0fa5c372fe8a1da76453ce146997c2f2ec5c0a
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup12](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104154/execution/node/956/log)

### History
* :broken_heart: [Build #104127](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104127/) failed ea9e086c9aa36ce3ca258e96e1d3e507f5d503ba
* :broken_heart: [Build #104111](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/) failed 7e05b927382e5ccfd03854e16b1844f243c951cd
* :broken_heart: [Build #103749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/) failed f27f49cd1a7aaa11f132b8695fb01b0a35e93d55

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/","number":"103749","commit":"f27f49cd1a7aaa11f132b8695fb01b0a35e93d55"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/","number":"104111","commit":"7e05b927382e5ccfd03854e16b1844f243c951cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104127/","number":"104127","commit":"ea9e086c9aa36ce3ca258e96e1d3e507f5d503ba"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104154/","number":"104154","commit":"ac0fa5c372fe8a1da76453ce146997c2f2ec5c0a"}]}
PIPELINE-->

cjcenizal commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104250/)
* Commit: 3797ce6150969ed646179aa9f5510e2ddfe6fdcb

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b058f7852b74fd22f396c367dc47cf4a526b4daa) | [after](https://github.com/elastic/kibana/commit/3797ce6150969ed646179aa9f5510e2ddfe6fdcb) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 136.4KB | 136.7KB | +295.0B |

### History
* :broken_heart: [Build #104154](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104154/) failed ac0fa5c372fe8a1da76453ce146997c2f2ec5c0a
* :broken_heart: [Build #104127](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104127/) failed ea9e086c9aa36ce3ca258e96e1d3e507f5d503ba
* :broken_heart: [Build #104111](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/) failed 7e05b927382e5ccfd03854e16b1844f243c951cd
* :broken_heart: [Build #103749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/) failed f27f49cd1a7aaa11f132b8695fb01b0a35e93d55

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/","number":"103749","commit":"f27f49cd1a7aaa11f132b8695fb01b0a35e93d55"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/","number":"104111","commit":"7e05b927382e5ccfd03854e16b1844f243c951cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104127/","number":"104127","commit":"ea9e086c9aa36ce3ca258e96e1d3e507f5d503ba"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104154/","number":"104154","commit":"ac0fa5c372fe8a1da76453ce146997c2f2ec5c0a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104250/","number":"104250","commit":"3797ce6150969ed646179aa9f5510e2ddfe6fdcb"}]}
PIPELINE-->

myasonik reviewed (approved):
LGTM! 🚀

cuff-links commented:
> Thanks for working on this @cuff-links! It's great to see more accessibility tests.
> 
> I left a few questions in the code. I also think you will need to add the test file to `x-pack/test/accessibility/config.ts` in order for the tests to run in CI.
> 
> ```
> require.resolve('./apps/upgrade_assistant'),
> ```

This has been resolved.

alisonelizabeth reviewed (approved):
Latest LGTM. Ran the tests successfully locally. Great work @cuff-links!

As a next step, I think it'd be great to add some a11y tests for when there are deprecation issues that need to be addressed (right now, only the empty prompt shows for the indices and cluster tab), but I think this is a great start!

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104691/)
* Commit: e5202150f1aea152f2a9c51b4bfdcf95368a7e87

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d201ed756f79bae54e1d4b03c44ecdf3ea3bdda6) | [after](https://github.com/elastic/kibana/commit/e5202150f1aea152f2a9c51b4bfdcf95368a7e87) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 136.4KB | 136.7KB | +295.0B |

### History
* :green_heart: [Build #104250](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104250/) succeeded 3797ce6150969ed646179aa9f5510e2ddfe6fdcb
* :broken_heart: [Build #104154](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104154/) failed ac0fa5c372fe8a1da76453ce146997c2f2ec5c0a
* :broken_heart: [Build #104127](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104127/) failed ea9e086c9aa36ce3ca258e96e1d3e507f5d503ba
* :broken_heart: [Build #104111](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/) failed 7e05b927382e5ccfd03854e16b1844f243c951cd
* :broken_heart: [Build #103749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/) failed f27f49cd1a7aaa11f132b8695fb01b0a35e93d55

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103749/","number":"103749","commit":"f27f49cd1a7aaa11f132b8695fb01b0a35e93d55"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/","number":"104111","commit":"7e05b927382e5ccfd03854e16b1844f243c951cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104127/","number":"104127","commit":"ea9e086c9aa36ce3ca258e96e1d3e507f5d503ba"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104154/","number":"104154","commit":"ac0fa5c372fe8a1da76453ce146997c2f2ec5c0a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104250/","number":"104250","commit":"3797ce6150969ed646179aa9f5510e2ddfe6fdcb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104691/","number":"104691","commit":"e5202150f1aea152f2a9c51b4bfdcf95368a7e87"}]}
PIPELINE-->

cuff-links commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104905/)
* Commit: 63a0022ee03055ac5ddde4cfc3f2bf6ccc3adcd9
* [Documentation Changes](https://kibana_90265.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/7a2b7550c962c5b6084f8f84e2dc9a2b5f3db0a2) | [after](https://github.com/elastic/kibana/commit/63a0022ee03055ac5ddde4cfc3f2bf6ccc3adcd9) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 136.4KB | 136.7KB | +295.0B |

### History
* :green_heart: [Build #104691](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104691/) succeeded e5202150f1aea152f2a9c51b4bfdcf95368a7e87
* :green_heart: [Build #104250](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104250/) succeeded 3797ce6150969ed646179aa9f5510e2ddfe6fdcb
* :broken_heart: [Build #104154](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104154/) failed ac0fa5c372fe8a1da76453ce146997c2f2ec5c0a
* :broken_heart: [Build #104127](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104127/) failed ea9e086c9aa36ce3ca258e96e1d3e507f5d503ba
* :broken_heart: [Build #104111](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/) failed 7e05b927382e5ccfd03854e16b1844f243c951cd

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104111/","number":"104111","commit":"7e05b927382e5ccfd03854e16b1844f243c951cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104127/","number":"104127","commit":"ea9e086c9aa36ce3ca258e96e1d3e507f5d503ba"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104154/","number":"104154","commit":"ac0fa5c372fe8a1da76453ce146997c2f2ec5c0a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104250/","number":"104250","commit":"3797ce6150969ed646179aa9f5510e2ddfe6fdcb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104691/","number":"104691","commit":"e5202150f1aea152f2a9c51b4bfdcf95368a7e87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104905/","number":"104905","commit":"63a0022ee03055ac5ddde4cfc3f2bf6ccc3adcd9"}]}
PIPELINE-->
